climada/hazard/centroids/centr.py

```diff
@@ -548,15 +553,19 @@ def union(self, *others):
         for cent in cent_list:
             if cent.meta and not cent.lat.any():
                 cent.set_meta_to_lat_lon()
-            if is_geom:
-                cent.set_geometry_points()
-                if not u_coord.equal_crs(cent.crs, self.crs):
-                        raise ValueError('Different CRS are not accepted.')
-            if is_area_pixel: cent.set_area_pixel()
-            if is_dist_coast: cent.set_dist_coast()
-            if is_on_land: cent.set_on_land()
-            if is_region_id: cent.set_region_id()
-            if is_elevation: cent.set_elevation()
+            cent.set_geometry_points()
+            if not u_coord.equal_crs(cent.crs, self.crs):
+                    raise ValueError('Different CRS are not accepted.')
+            if is_area_pixel and not cent.area_pixel.any():
+                cent.set_area_pixel()
+            if is_dist_coast and not cent.dist_coast.any():
+                cent.set_dist_coast()
+            if is_on_land and not len(cent.on_land) > 0:
+                cent.set_on_land()
+            if is_region_id and not cent.region_id.any():
+                cent.set_region_id()
+            if is_elevation and not cent.elevation.any():
+                cent.set_elevation()
```

Comment:
since none of these values are required to remove duplicates, is there a smart way to calculate them _after_ everything is combined and duplicates are removed? for large sets of centroids with a lot of overlapping points, it could save time.

this might be a bit of an edge case that we don't need to worry about: i don't know when you'd want to combine _N_ centroids objects where 1 < _x_ << _N_ of the objects have this calculated already.

Reply:
I tried to just put it at the end, and then realized why I put it where it is now. The `centroids.remove_duplicate_points` will select from all the numpy arrays. Hence, if one of them is not defined, it fails. 
As a compromise, the method only computes the missing values if they are missing :D

Reply:
I'm trying to find where that happens. Is it in `Centroids.select()`, lines 814-821 in centr.py?

This isn't very important though - in the edge case I'm describing the the user can plan around accidental large calculations, since the documentation is clear about what the code does.